### PR TITLE
[LS-007-B] Implementation finished, merge into Dev Branch [PL,RT]

### PR DESCRIPTION
**Pull request by tom-reith**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -55,5 +59,5 @@ dependencies {
     // optional - Test helpers
     testImplementation "androidx.room:room-testing:$room_version"
 
-    androidTestImplementation 'androidx.test:rules:1.1.0'
```
Comment (hydrogen-1) [inline app/build.gradle:58]: don't remove this, makes tests not run

File: app/src/androidTest/java/com/example/loginsesame/TestApplicationLogin.kt [hunk 2]
```diff
@@ -56,7 +83,7 @@ class TestApplicationLogin {
     }
 
     @Test
-    fun userPressesBackButton() {
+    fun userClicksBackButton() {
```
Comment (hydrogen-1) [inline app/src/androidTest/java/com/example/loginsesame/TestApplicationLogin.kt:97]: Test will not run if soft-keyboard is visible, close keyboard with espresso before pressing back

**Review by hydrogen-1 — CHANGES_REQUESTED** [review]
Change build.gradle to include ```androidTestImplementation 'androidx.test:rules:1.1.0'``` and change userClicksBackButton test to close soft-keyboard before pressing back

File: app/build.gradle (hunk 1 above)
Comment (tom-reith) [reply app/build.gradle:58]: we just moved it to the top to not make it stand out anymore

File: app/src/androidTest/java/com/example/loginsesame/TestApplicationLogin.kt (hunk 2 above)
Comment (tom-reith) [reply app/src/androidTest/java/com/example/loginsesame/TestApplicationLogin.kt:97]: this test doesn't open the keyboard.
It just presses the hardware back button which is always visible. I have added it to the other tests though

**Comment by TPichler15** [comment]
Pull Request accepted

**Pull request merged**